src/main/java/org/springframework/data/util/ClassTypeInformation.java

```diff
@@ -70,7 +74,7 @@ public class ClassTypeInformation<S> extends TypeDiscoverer<S> {
 	 */
 	@Deprecated
 	public static <S> ClassTypeInformation<S> from(Class<S> type) {
-		return from(ResolvableType.forClass(type));
+		return from(resolvableTypeCache.get(type));
```

Comment:
👍 - I just realised the `ResolvableType` internal cache does not cover this part.